addons/pkg-maven/common/src/main/java/org/commonjava/indy/pkg/maven/content/index/MavenIndexingStrategy.java

```diff
@@ -0,0 +1,48 @@
+package org.commonjava.indy.pkg.maven.content.index;
+
+import org.commonjava.indy.content.index.PackageIndexingStrategy;
+import org.commonjava.indy.util.PathUtils;
+import org.commonjava.maven.galley.model.SpecialPathInfo;
+import org.commonjava.maven.galley.spi.io.SpecialPathManager;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import static org.commonjava.indy.pkg.PackageTypeConstants.PKG_TYPE_MAVEN;
+
+@ApplicationScoped
+@Named
+public class MavenIndexingStrategy
```

Comment:
@ligangty This is where I moved the main part of your directory-level indexing logic, and made it specific to maven repositories.